index.js

```diff
@@ -216,17 +218,33 @@ async function installBundler(bundlerVersionInput, lockFile, platform, rubyPrefi
     bundlerVersion = '1'
   }
 
+  const gem = path.join(rubyPrefix, 'bin', 'gem')
+
   if ((engine === 'ruby' || engine === 'truffleruby') && common.isHeadVersion(rubyVersion) && bundlerVersion === '2') {
     console.log(`Using Bundler 2 shipped with ${engine}-${rubyVersion}`)
   } else if (engine === 'truffleruby' && !common.isHeadVersion(rubyVersion) && bundlerVersion === '1') {
     console.log(`Using Bundler 1 shipped with ${engine}`)
   } else {
-    const gem = path.join(rubyPrefix, 'bin', 'gem')
     await exec.exec(gem, ['install', 'bundler', '-v', `~> ${bundlerVersion}`, '--no-document'])
   }
+
+  let bundlerSpecification = ''
+
+  const options = {
+    listeners: {
+      stdout: (data) => bundlerSpecification += data.toString()
+    }
+  }
+
+  await exec.exec(gem, ['specification', 'bundler', '-v', `~> ${bundlerVersion}`, 'version'], options)
```

Comment:
This would take extra time for all executions of the action, for a corner case, so that's not ideal.
But I think there is a better solution, see my other comments.